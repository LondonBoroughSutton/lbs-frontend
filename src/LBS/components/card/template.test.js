/**
 * @jest-environment jsdom
 */
/* eslint-env jest */

const axe = require('../../../../lib/axe-helper')

const { render, getExamples, htmlWithClassName } = require('../../../../lib/jest-helpers')

const examples = getExamples('card')

describe('Card', () => {
  describe('default example', () => {
    it('passes accessibility tests', async () => {
      const $ = render('card', examples.default)
      const results = await axe($.html())
      expect(results).toHaveNoViolations()
    })

    it('has correct nesting order', () => {
      const $ = render('card', examples.default)
      const $component = $('.lbs-card > h2')
      expect($component.length).toBeTruthy()
    })

    it('matches snapshot', () => {
      const $ = render('card', examples.default)
      expect(htmlWithClassName($, '.lbs-card')).toMatchSnapshot()
    })
  })

  describe('link in header example', () => {
    it('has a link in the header', () => {
      const $ = render('card', examples['card with link in header'])
      const $component = $('.lbs-card > h2 a')
      expect($component.length).toBeTruthy()
    })
  })

  describe('alt background version', () => {
    it('has expected class', () => {
      const $ = render('card', examples['card with background colour'])
      const $component = $('.lbs-card--alt')
      expect($component.length).toBeTruthy()
    })
  })

  describe('navigation list examples', () => {
    it('passes accessibility tests', async () => {
      const $ = render('card', examples['card with links'])

      const results = await axe($.html())
      expect(results).toHaveNoViolations()
    })

    it('contains an unordered list', () => {
      const $ = render('card', examples['card with links'])

      const $component = $('.lbs-card ul')
      expect($component.length).toBeTruthy()
    })

    it('contains only 3 list items', () => {
      const $ = render('card', examples['card with links'])
      const $component = $('.lbs-card ul li')
      expect($component.length).toEqual(3)
    })

    it('matches snapshot', () => {
      const $ = render('card', examples['card with links'])
      expect($).toMatchSnapshot()
    })
  })

  describe('navigation list examples', () => {
    it('passes accessibility tests', async () => {
      const $ = render('card', examples['card with increased limit of links and `show more` feature'])

      const results = await axe($.html())
      expect(results).toHaveNoViolations()
    })

    it('contains an unordered list', () => {
      const $ = render('card', examples['card with increased limit of links and `show more` feature'])
      const $component = $('.lbs-card ul')
      expect($component.length).toBeTruthy()
    })

    it('contains only 3 list items without the `js__is-hidden` class', () => {
      const $ = render('card', examples['card with increased limit of links and `show more` feature'])
      const $component = $('.lbs-card ul li:not(.js__is-hidden)')
      expect($component.length).toEqual(3)
    })

    it('matches snapshot', () => {
      const $ = render('card', examples['card with links'])
      expect($).toMatchSnapshot()
    })
  })
})
