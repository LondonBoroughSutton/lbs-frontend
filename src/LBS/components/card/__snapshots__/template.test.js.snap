// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card default example matches snapshot 1`] = `
<div class="lbs-card   ">
  This is a description
</div>
`;

exports[`Card navigation list examples matches snapshot 1`] = `[Function]`;

exports[`Card navigation list examples matches snapshot 2`] = `[Function]`;

exports[`Card popular items matches snapshot 1`] = `[Function]`;
